flow/src/test/java/com/sequenceiq/flow/core/Flow2HandlerTest.java

```diff
@@ -564,6 +565,7 @@ public void testRestartFlow() throws TransactionExecutionException {
         Payload payload = new TestPayload(STACK_ID);
         flowLog.setPayloadType(ClassValue.of(TestPayload.class));
         flowLog.setPayload(JsonWriter.objectToJson(payload));
+        flowLog.setPayloadJackson(JsonUtil.writeValueAsStringSilent(payload));
```

Comment:
If I get it right, this will only verify the `fl.getPayloadJackson() != null` branch (relying on `JsonUtil.readValueWithJsonIoFallback()`), but skip the `else` branch (using `JsonReader.jsonToJava()`).

Reply:
covered both cases